src/Npgsql/TypeHandling/NpgsqlTypeHandler.cs

```diff
@@ -96,6 +96,21 @@ internal async ValueTask<TAny> ReadWithLength<TAny>(NpgsqlReadBuffer buf, bool a
                    : await Read<TAny>(buf, len, async, fieldDescription);
         }
 
+        /// <summary>
+        /// Reads a value from the buffer, assuming our read position is at the value's preceding length.
+        /// If the length is -1 (null), this method will return <see langword="null"/>.
+        /// </summary>
+        internal async ValueTask<object?> ReadNullableWithLength<TAny>(NpgsqlReadBuffer buf, bool async, FieldDescription? fieldDescription = null)
```

Comment:
It seems like this is only used by array handler - seems like it should just be a local method there, no?

Reply:
Since it was only used in one place and consisted of three expressions I decided to inline it.